cmds/dump.c

```diff
@@ -1111,6 +1115,89 @@ static void print_flame_footer(struct uftrace_dump_ops *ops,
 	graph_remove_task();
 }
 
+/* to graphviz support */
+static struct uftrace_graph graphviz_graph = {
+	.root.head     = LIST_HEAD_INIT(graphviz_graph.root.head),
+	.special_nodes = LIST_HEAD_INIT(graphviz_graph.special_nodes),
+};
+
+static void print_graphviz_header(struct uftrace_dump_ops *ops,
+				  struct ftrace_file_handle *handle,
+				  struct opts *opts)
+{
+	pr_out("# version\":\"uftrace %s\",\n", UFTRACE_VERSION);
+
+	if (handle->hdr.info_mask & (1UL << CMDLINE))
+		pr_out("# command_line \"%s\"\n", handle->info.cmdline);
+
+	pr_out("digraph \"");
+	pr_out("%s", (&handle->info)->exename);
+	pr_out("\" { \n");
+	pr_out("\n\t# Attributes \n");
+	pr_out("\tsplines=ortho;\n");
+	pr_out("\tconcentrate=true;\n");
+	pr_out("\tnode [shape=\"rect\",fontsize=\"7\",style=\"filled\"];\n");
+	pr_out("\tedge [fontsize=\"7\"];\n\n");
+
+	graph_init_callbacks(NULL, NULL, NULL, ops);
+}
+
+static void print_graphviz_task_rstack(struct uftrace_dump_ops *ops,
+				       struct ftrace_task_handle *task,
+				       char *name)
+{
+	struct uftrace_record *frs = task->rstack;
+	struct uftrace_task_graph *graph;
+
+	graph = graph_get_task(task, sizeof(*graph));
+
+	graph->graph = &graphviz_graph;
+	graphviz_graph.sess = find_task_session(&task->h->sessions,
+			task->tid, frs->time);
```

Comment:
one more indentation nit: the argument line should start from the same position with the above line.